observations/pyspark_obs_domain_batch.py

```diff
@@ -91,34 +96,35 @@
 debug_logBackuphandler = TimedRotatingFileHandler(f"{file_name_for_debug_log}",when="w0",backupCount=1)
 infoLogger.addHandler(debug_logHandler)
 infoLogger.addHandler(debug_logBackuphandler)
-
+duplicate_checker = None
 #Check for duplicate
 data_fixer = None
 datasource_name = json.loads(config.get("DRUID","ml_distinctCnt_obs_domain_spec"))["spec"]["dataSchema"]["dataSource"]
 try:
-    with open(f'{config.get("COMMON","obs_tracker_path_domain_batch")}', 'r', newline='') as file:
-        reader = csv.DictReader(file)
-        for row in reader:
-            if row['datasource'] == datasource_name:
-                if row['task_created_date'] != str(datetime.datetime.now().date()):
-                    # Check: Daily run - Date is not duplicate
-                    duplicate_checker = False
-                else: 
-                    druid_id = row['task_id']
-                    druid_status = requests.get(f'{config.get("DRUID", "batch_url")}/{druid_id}/status')
-                    druid_status.raise_for_status()
-                    ingest_status = druid_status.json()["status"]["status"]
-                    if ingest_status == 'SUCCESS':
-                        # Check: Date is duplicate
-                        duplicate_checker = True
-                        bot.api_call("chat.postMessage",channel=config.get("SLACK","channel"),text=f"ABORT: 'Duplicate-run' for {datasource_name}")
-                    else:
-                        # Check: Date duplicate but ingestion didn't get processed
-                        duplicate_checker = False
-                        data_fixer = True
-            else:
-                # Check: New task_id or No task_id
-                duplicate_checker = False
+   # construct query for log 
+   today = str(current_date)
+   query = {"dataSource" : datasource_name , "taskCreatedDate" : today}
+
+   logCheck = getLogs(query)
+   if not logCheck['duplicateChecker']:
+     duplicate_checker = logCheck['duplicateChecker']
+   else:
+     duplicate_checker = logCheck['duplicateChecker']
+     if logCheck['dataFixer']:
+        data_fixer = True
+     else:
+        druid_id = logCheck['response']['taskId']
+        druid_status = requests.get(f'{config.get("DRUID", "batch_url")}/{druid_id}/status')
```

Comment:
@adithyadinesh0412 we can take this from constatus, If we are using this in multiple places

Reply:
Yes @rakeshSgr  , The url is taken from config file (as it can change with envs). druid id is a dynamic value.